### DISCUSSION TITLE
Add authentication to boost UI and allow direct login via URL

### DISCUSSION THREAD

**Discussion by LexLuthr** [filecoin-project/boost#427] [Ideas]
Can we allow direct login via URL? Let's say boost is running on x.x.x.x and exposed port for UI is 8080 then https://x.x.x.x:8080 should allow login to the UI. This would enable remote monitoring for deals.

We would need to enable authentication for the UI to allow direct logins.

Even if direct login is not possible to implement for security reason, can authentication still be integrated as an extra security layer?

**Comment by dirkmc** [comment]
We suggest that rather than building authentication into Boost, users can add an authentication layer themselves using a reverse proxy such as nginx

**Reply by cryptowhizzard** [reply]
Haproxy also does the trick, including authentification.

**Comment by LexLuthr** [comment]
@sbudo FYI ^

**Comment by LexLuthr** [comment]
A few potential use cases (but then again, its a “nice to have” feature):

1. For larger businesses who need to protect the UI internally (I don’t want all my employees to be able to access the site). Now that it works without ssh tunnel..
2. There is also a case of admins who aren’t potentially proficient in Linux but have been asked to manage deals through the UI
3. Mobile devices from which ssh tunnel is unavailable (I can check my deals status from the beach scenario)
